### PR TITLE
Ft product search

### LINKED ISSUE #70: Feat: Buyer should be able to search a product
**Description**

The buyer should be able to search inside the site.

**Acceptance Criteria**

- [ ] The buyer can use name to search for product.
- [ ] Search results are reliable and relevant from the products record.
- [ ] Filtering and sorting options are supported by the search functionality.
- [ ] The implementation of pagination helps managing big result sets.

**Developer Notes**

- Provide a search feature for the product catalog service .
- Create API endpoints to manage requests for the product searches.
- Make use of effective querying and indexing strategies to provide the best search results.

**Related Issues**

#49 
#48 

**Definition of done**

- The creation and integration of API endpoints enables them to fulfill buyer requests for the product searches.
- The buyer can use name to search for the product.
- The search includes options for sorting and filtering.
- In order to manage big result sets, pagination is used.

### PR DESCRIPTION
**Pull request by GSinseswa721**
**Description**

This PR is all about the implementation of the product search, so that the buyer could be able to search inside the e-commerce.

**Functionality**

- Searching the product by using name.
- Using filtering and sorting to retrieve data from database.
- Implementation of pagination for managing big result.

### CODE REVIEW

**Comment by faid-terence** [comment]
@GSinseswa721 Please before starting we need to see your implementation plan !

**Review by Calebgisa72 — APPROVED** [review]

**Review by maxCastro1 — APPROVED** [review]

File: src/routes/ProductRoutes.ts
Comment (faid-terence) [inline src/routes/ProductRoutes.ts:1]: @GSinseswa721 
Remove as any on middlwares

File: src/utils/auth.ts
Comment (faid-terence) [inline src/utils/auth.ts:1]: @GSinseswa721 
Remove console.log("......................")s in the project

**Review by faid-terence — CHANGES_REQUESTED** [review]

**Comment by faid-terence** [comment]
@GSinseswa721 
Progress on this PR ?

**Comment by faid-terence** [comment]
@GSinseswa721 Resolve the conflicts and increase the coverage as well

**Review by Ndevu12 — APPROVED** [review]

**Review by UwicyezaG — APPROVED** [review]

**Review by elijahladdie — APPROVED** [review]

**Review by faid-terence — APPROVED** [review]
Nice Work @GSinseswa721 !

**Pull request merged**